From ceee9e3cec4464a68b07583bf209d3dfa30ef065 Mon Sep 17 00:00:00 2001
From: "flinux@vongellhorn.ch" <flinux@vongellhorn.ch>
Date: Tue, 14 May 2019 12:09:51 +0200
Subject: [PATCH 1/1] Packport in the new location for knsrc files

search in the new knsrc location, still fallback to the old one for
compatibility

https://phabricator.kde.org/D21183
---
 src/kpackage-install-handlers/kns/main.cpp | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/src/kpackage-install-handlers/kns/main.cpp b/src/kpackage-install-handlers/kns/main.cpp
index 798b13c..b42fc8d 100644
--- a/src/kpackage-install-handlers/kns/main.cpp
+++ b/src/kpackage-install-handlers/kns/main.cpp
@@ -24,6 +24,7 @@
 #include <QTimer>
 #include <QDebug>
 #include <QStandardPaths>
+#include <QFile>
 
 #include <KLocalizedString>
 
@@ -46,7 +47,15 @@ int main(int argc, char** argv)
     Q_ASSERT(url.isValid());
     Q_ASSERT(url.scheme() == QLatin1String("kns"));
 
-    const auto knsname = QStandardPaths::locate(QStandardPaths::GenericConfigLocation, url.host());
+    QString knsname;
+    for (const auto &location : KNSCore::Engine::configSearchLocations()) {
+        QString candidate = location + QLatin1Char('/') + url.host();
+        if (QFile::exists(candidate)) {
+            knsname = candidate;
+            break;
+        }
+    }
+    
     if (knsname.isEmpty()) {
         qWarning() << "couldn't find knsrc file for" << url.host();
         return 1;
-- 
2.21.0

